test/CMakeLists.txt

```diff
@@ -4,7 +4,7 @@ if(UNIX)
 endif()
 
 IF("${CMAKE_CXX_COMPILER_ID}" MATCHES "(Apple)?[Cc]lang")
-    set(CMAKE_CXX_FLAGS "-Wunused-command-line-argument -std=c++20 -Wall -Wextra -Wfatal-errors -Werror=return-type -g -O0 -fno-pie")
+    set(CMAKE_CXX_FLAGS "-Wunused-command-line-argument -std=c++20 -Wall -Wextra -Wfatal-errors -Werror=return-type -g -O0 -fpie")
```

Comment:
Why this change?

Reply:
Inside the Dockerfile i uploaded there is an option to enable compilation with Clang, which in some cases gives detailed compile diagnostics. I also installed clang-tidy within the image to help with formatting and code style later on.

To enable Clang compilation, you can unsilence the environment variables in the bottom of the Dockerfile:
...
# # (Optional) Set Clang as the default compiler
# ENV CC=/usr/bin/clang
# ENV CXX=/usr/bin/clang++
...

and make a clean rebuild.

When building with Clang this error occurs:

`Linking CXX executable ../../bin/test_cases_dynamic
/usr/bin/ld: CMakeFiles/test_cases_dynamic.dir/test_discrete_lti_siso.cpp.o: relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/ld: failed to set dynamic section sizes: bad value
clang: fatal error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test/CMakeFiles/test_cases_dynamic.dir/build.make:148: ../bin/test_cases_dynamic] Error 1
make[1]: *** [CMakeFiles/Makefile2:912: test/CMakeFiles/test_cases_dynamic.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....`

Reply:
Gotcha, that's fine!